_common/MarkdownHost.php

```diff
@@ -41,14 +41,19 @@ function __construct($file) {
       // source: https://yaml.org/spec/1.2/spec.html#id2760395
       // source: https://pandoc.org/MANUAL.html#extension-yaml_metadata_block
       //
-      if(preg_match('/^---\n(.+)\n---\n(.+)/s', $contents, $match)) {
+      if(preg_match('/^---\n(([a-z0-9_-]+:.+\n)+)---(\n)?((.|\n)*)$/i', $contents, $match)) {
         $metadata = $match[1];
-        $contents = $match[2];
+        $contents = $match[4];
       } else {
         $metadata = 'title: untitled';
       }
 
-      if(preg_match('/^title: (.+)/', $metadata, $match)) {
+      if(preg_match('/^redirect: (.+)/m', $metadata, $match)) {
```

Comment:
In keymanapp/help.keyman.com#337 the header at l.34 was updated for title and redirect.